src/bors/handlers/review.rs

```diff
@@ -25,7 +25,7 @@ pub(super) async fn command_approve(
     priority: Option<u32>,
 ) -> anyhow::Result<()> {
     tracing::info!("Approving PR {}", pr.number);
-    if !sufficient_approve_permission(repo_state.clone(), author) {
+    if !has_review_or_delegated_permission(&repo_state, author, pr, &db).await? {
```

Comment:
```suggestion
    if !has_approve_permissions(&repo_state, author, pr, &db).await? {
```
I think that name is clearer (and doesn't leak implementation details :) ).